### ISSUE TITLE
Improve key bindings in custom server selection on login [Papercut]

### ISSUE THREAD

**Issue by spaetz** [element-hq/element-web#31787]
#### What would you like to do?
After entering my non-matrix.org server to log into (I dismiss that dialog by hitting enter), I would like to just hit enter one more time on the below confirmation dialog (sorry German), to proceed. 
<img width="726" height="290" alt="Image" src="https://github.com/user-attachments/assets/d4aeb808-8230-4aed-9853-fb1710ff377c" />

Instead, when I hit enter on that dialog I am thrown back into the "enter your server name" dialog. It would be great if I could just confirm my choice (by keyboard), instead of clicking the ONLY real button there ('Continue') by mouse.

#### Why would you like to do it?

Hitting enter should chose the only real button in a dialog, instead of leading to a side-quest.

#### How would you like to achieve it?

Push the blinkenlight!

**Comment by t3chguy** [comment]
> Instead, when I hit enter on that dialog I am thrown back into the "enter your server name" dialog.

This is as per Accessibility best practices, for a user relying on a screen reader this can be quite important, when a modal/dialog/menu is closed focus should be returned back to where it was before the modal/dialog/menu was open.

**Comment by spaetz** [comment]
Ahh, that makes sense. The question is why the default focus is on changing the server name  in the first place when there is such a prominent, default-looking "continue" button on the dialog. I will let you decide, I can leave with either solution. I just fall into that trap every single time, I log into EW :-).

**Comment by t3chguy** [comment]
> The question is why the default focus is on changing the server name in the first place

Not sure what you mean by this, there is no default focus on that page as far as I can see.

<img width="1714" height="1012" alt="Image" src="https://github.com/user-attachments/assets/a61517b4-9129-4edc-bb2e-cda3d877ff2b" />

This shows the focus is on `body`

https://github.com/user-attachments/assets/50358911-ca72-41dc-811f-49bd9f702835

And as I tab around it goes

+ Language selector
+ `i`
+ `matrix.org`
+ Edit
+ Continue

Also the `Continue` button is rendered dynamically, after we contact the server (matrix.org in the case of my test) to know what auth methods it supports. Moving focus to an element which does not render immediately could cause usability issues, especially on a slower connection, as you'd be tabbing around then your focus would jump to `Continue` because that just loaded in.

**Comment by spaetz** [comment]
OK, thanks for testing that and the additional explanations, I had not tried to hit ENTER in the initial screen. Mea culpa.

**Comment by dbkr** [comment]
Sounds like there's nothing to do here so I'm going to close this.
